app/Services/ShareWithAFriendService.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+
+namespace App\Services;
+
+
+use App\Mail\ShareWithAFriendMail;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Mail;
+
+class ShareWithAFriendService
+{
+    public function inviteFriend(array $data)
+    {
+        $user = Auth::user();
```

Comment:
@enwemasor1995 Do not call Auth::User() in the service. Get the user from the GraphQL resolver. Check the mutation for `WithdrawFromContributionPlanToWallet` to see how to inject the user into the args for the graphql class, you can then access it as $args['user'] and pass that to the service